### DISCUSSION TITLE
Acknowledgement for Incoming HL7 Message

### DISCUSSION THREAD

**Discussion by snarayanan99** [nextgenhealthcare/connect#6349] [Q&A]
I am receiving an incoming MDM-T02 HL7 message to the Mirth Source Connector. I need to send an ACK to the client before I begin processing of the message
I set the Source Queue setting to "ON" and Response to "Auto Generate (before Processing)" in the Source Connector
The Channel as such works fine and processing is occurring as expected
But I am unable to see the ACK message in the Mirth logs or in the Mirth administrator Server  logs 
How do I confirm that an ACK is going out for the incoming HL7 message and how do I view it in the logs
Your inputs on this are much appreciated

**Comment by kirbykn2** [comment]
On the source, click on the 'Response' radio button.
[image: image.png]

On Tue, Nov 12, 2024 at 8:14 AM snarayanan99 ***@***.***>
wrote:

> I am receiving an incoming MDM-T02 HL7 message to the Mirth Source
> Connector. I need to send an ACK to the client before I begin processing of
> the message
> I set the Source Queue setting to "ON" and Response to "Auto Generate
> (before Processing)" in the Source Connector
> The Channel as such works fine and processing is occurring as expected
> But I am unable to see the ACK message in the Mirth logs or in the Mirth
> administrator Server logs
> How do I confirm that an ACK is going out for the incoming HL7 message and
> how do I view it in the logs
> Your inputs on this are much appreciated
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/nextgenhealthcare/connect/discussions/6349>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APRXWDZTTS7GI2MQ4DAIFX32AH5MHAVCNFSM6AAAAABRUBF2Z6VHI2DSMVQWIX3LMV43ERDJONRXK43TNFXW4OZXGQ3DSOJVGU>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Best,

Kirby Knight
| 231.735.4650 | ***@***.***

**Reply by snarayanan99** [reply]
I dont see a Response radio button. I see only a Response checkbox where I chose "Auto-generate (before processing". I have attached my Source snapshot for your reference
![image](https://github.com/user-attachments/assets/54561a07-a8db-4a09-ad0d-bebe6464ebc5)

**Reply by pacmano1** [reply]
In the message viewer you will see the response when you get a message.

**Reply by snarayanan99** [reply]
I am parsing an incoming HL7 message as an attachment. The incoming message size could be as big as 40MB. The Attachment javascript code is as below
var attachment = addAttachment(message ,"text/plain" , false);
return attachment.getAttachmentId();
This does not send out an acknowledgement  even though settings on Source Connector are - Source Queue setting is "ON" and Response is "Auto Generate (before Processing)"

**Comment by kirbykn2** [comment]
You need to be in the dashboard view, not the channel configuration, to see
messages and responses.

[image: image.png]

On Tue, Nov 12, 2024 at 9:29 AM pacmano1 ***@***.***> wrote:

> In the message viewer you will see the response when you get a message.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/nextgenhealthcare/connect/discussions/6349#discussioncomment-11227375>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APRXWD4FAMNJQ3FRXXGRZ232AIGF5AVCNFSM6AAAAABRUBF2Z6VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCMRSG4ZTONI>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***
> com>
>


-- 
Best,

Kirby Knight
| 231.735.4650 | ***@***.***

**Reply by snarayanan99** [reply]
I am parsing an incoming HL7 message as an attachment. The incoming message size could be as big as 40MB. The Attachment javascript code is as below
var attachment = addAttachment(message ,"text/plain" , false);
return attachment.getAttachmentId();
This does not send out an acknowledgement even though settings on Source Connector are - Source Queue setting is "ON" and Response is "Auto Generate (before Processing)"

**Reply by pacmano1** [reply]
Firstly, it doesn't make sense usually to store the entire message as an attachment, it does however make sense to use a regex to grab the part of the message that is large.  Ignore the channel name on the screenshot.

![image](https://github.com/user-attachments/assets/22a5b337-a5de-4944-b531-7a7c03329a86)

Also,  you are duplicating already available capability in Mirth.   Read https://forums.mirthproject.io/articles/175046-using-attachments

**Reply by snarayanan99** [reply]
Thanks for the suggestion. I shall try the regex expression. Also is there a way to save the auto generated response to the channelMap, so that I can retrieve it in the Database writer destination and persist the auto acknowledgement message in the database

**Reply by snarayanan99** [reply]
My requirement is to send an ACK before processing a 40 MB incoming HL7 message. The same ACK needs to be written to a channelMap so that I can persist that outgoing ACK to the database later in my Database Writer destination

**Reply by pacmano1** [reply]
These are all possible and fundamental functions of Mirth.  I'd advise you read the manual, also review the wealth of information on the mirth forums and take some formal training.  https://www.udemy.com/course/learning-mirth-connect/ is inexpensive and thorough.  

Note that folks are happy to help here when you are having problems with code and post screenshots and samples.  but when you just ask without providing your own effort, typically you don't get a lot of responses.

**Reply by snarayanan99** [reply]
Thanks for your suggestion on the Udemy course.. I did not find a lot of information on the forums on the specific use case that I have. That is why I had posted after trying all possible options that I could think of

**Reply by pacmano1** [reply]
https://forums.mirthproject.io/forum/mirth-connect/support/175532-capture-ack-and-store-it-in-db is your question on the ACK.
